test/core/pipeline/features/test_run.py

```diff
@@ -1065,7 +1065,7 @@ def fake_generator_run(self, generation_kwargs: Optional[Dict[str, Any]] = None,
         ]
     )
 
-    pipe = pipeline_class(max_runs_per_component=1)
+    pipe = pipeline_class(max_runs_per_component=2)
```

Comment:
These are now the expected correct number. This wasn't caught before because the order of the if checks was incorrect.